RHDEVDOCS-3111: Update monitoring flows and capture customizing metrics

smunje1 opened the pull request:
[RHDEVDOCS-3111](https://issues.redhat.com/browse/RHDEVDOCS-3111): Update monitoring flows and capture customizing metrics
Aligned team: Dev Tools
OCP version for cherry-picking: enterprise-4.11
JIRA issues: [RHDEVDOCS-3111](https://issues.redhat.com/browse/RHDEVDOCS-3111)
Preview pages: [Download to see the preview in your browser_1](https://drive.google.com/file/d/1VxkJcQ2igBzBhrKsE-Arb4rqUe8wPCz3/view?usp=sharing)
[Download to see the preview in your browser_2](https://drive.google.com/file/d/1wjw4AemFpDxL-aIW49k5H425jNQ11pfm/view?usp=sharing)
SME Review: @vikram-raj 
QE review: @sanketpathak
Peer-review

Note: As per the discussion with SME @vikram-raj, the Affects Version/s is Openshift 4.9 for this Jira (https://issues.redhat.com/browse/RHDEVDOCS-3111) . Therefore, I have raised the PR pointing to 4.11 (main) and we need to cherry-pick it till 4.9.

smunje1 commented:
Hello [Vikram Raj](https://issues.redhat.com/secure/ViewProfile.jspa?name=viraj-1) 
Could you please help us with the SME review for this Jira?

Thank you!

smunje1 commented:
Note: We are covering only the epic https://issues.redhat.com/browse/ODC-3698 in this Jira. Please see the [thread](https://coreos.slack.com/archives/CHGU4P8UU/p1658804326896729).

vikram-raj commented on modules/odc-monitoring-your-project-metrics.adoc:
```diff
@@ -11,19 +11,25 @@ After you create applications in your project and deploy them, you can use the *
 Use the following options to see further details:
 
-** Select a workload from the *All Workloads* list to see the filtered metrics for the selected workload.
+** Select a dashboard in the *Dashboard* list to see the filtered metrics. All dashboards produce additional sub-menus when selected, except *Kubernetes / Compute Resources / Namespace (Pods)*.
```
```suggestion
** Select a dashboard in the *Dashboards* list to see the filtered metrics. All dashboards produce additional sub-menus when selected, except *Kubernetes / Compute Resources / Namespace (Pods)*.
```

vikram-raj reviewed (commented):
LGTM

smunje1 commented:
Hello @sanketpathak 
Could you please help us with the QE review for this PR?

Thank you!

sanketpathak commented:
LGTM

sanketpathak commented:
LGTM

Srivaralakshmi commented on modules/odc-monitoring-your-project-metrics.adoc:
```diff
@@ -11,19 +11,25 @@ After you create applications in your project and deploy them, you can use the *
-** Click on any of the graphs displayed to see the details for that particular metric in the *Metrics*  page.
+** Click *Inspect*, which is in the upper-right corner of every graph, to see any particular graph details in the *Metrics* tab.
 
 * Use the *Metrics* tab to query for the required project metric.
```
Suggestion for L59:
. Optional: Use the *Events* tab to see the events for your project.

Srivaralakshmi commented on monitoring/reviewing-monitoring-dashboards.adoc:
```diff
@@ -20,15 +20,15 @@ In the *Administrator* perspective you can access dashboards for core {product-t
 image::monitoring-dashboard-administrator.png[]
 
-In the *Developer* perspective you can access dashboards that provide the following statistics for a selected project:
+In the *Developer* perspective, you can access Kubernetes compute resources dashboards that provide the following statistics for a selected project:
```
```suggestion
Use the *Developer* perspective to access Kubernetes compute resources dashboards that provide the following application metrics for a selected project:
```

Trying to be consistent with the usage of words.

Srivaralakshmi replied:
Suggestion for L11: 
Use the *Administrator* perspective to access dashboards for the core {product-title} components, including the following <items/components>:

Suggest using a complete sentence for the lead-in sentence to reduce ambiguity and support translation. Refer to: https://redhat-documentation.github.io/supplementary-style-guide/#lead-in-sentences

smunje1 replied:
This seems to be a better lead-in sentence, "Use the *Administrator* perspective to access dashboards for the core {product-title} components, including the following items:"

Cc @vikram-raj

Srivaralakshmi reviewed (changes requested):
@smunje1 PTAL at the suggestions. Thanks!

Srivaralakshmi commented:
I consulted the OpenShift checklist when conducting this peer review: https://docs.google.com/document/d/1WN7k72PKxcPA__erp5TK-CefmS_sVwE3r40PvpmNk3E/edit#heading=h.mgz8v4w1kg86

smunje1 commented on applications/odc-monitoring-project-and-application-metrics-using-developer-perspective.adoc:
```diff
@@ -19,5 +19,6 @@ include::modules/odc-monitoring-your-project-metrics.adoc[leveloffset=+1]
 include::modules/odc-monitoring-your-application-metrics.adoc[leveloffset=+1]
 
 [role="_additional-resources"]
+[id="additional-resources-odc-monitoring-project-and-application-metrics-using-developer-perspective"]
```
Done.

smunje1 commented on modules/monitoring-reviewing-monitoring-dashboards-developer.adoc:
```diff
@@ -16,10 +16,15 @@ In the Developer perspective, you can view dashboards relating to a selected pro
-. Choose a project in the *Project:* list.
-
-. Choose a workload in the *All Workloads* list.
+. Select a project in the *Project:* list.
```
Done.

smunje1 commented on modules/monitoring-reviewing-monitoring-dashboards-developer.adoc:
```diff
@@ -6,7 +6,7 @@
 = Reviewing monitoring dashboards as a developer
 
-In the Developer perspective, you can view dashboards relating to a selected project. You must have access to monitor a project to view dashboard information for it.
+In the Developer perspective, you can view Kubernetes compute resources dashboards relating to a selected project. You must have access to monitor a project to view dashboard information for it.
```
Hello @vikram-raj. Could you please share your feedback? Thank you!

vikram-raj replied:
Agree with @Srivaralakshmi. Rewriting the sentence doesn't change the technical meaning. And as we mentioned in the prerequisite we do not need to mention it again.

smunje1 replied:
Thanks, @vikram-raj  for your input. Rephrased as suggested and removed the latter sentence (as the same info is already covered in the prerequisite section).

smunje1 commented on modules/monitoring-reviewing-monitoring-dashboards-developer.adoc:
```diff
@@ -16,10 +16,15 @@ In the Developer perspective, you can view dashboards relating to a selected pro
-. Choose a workload in the *All Workloads* list.
+. Select a project in the *Project:* list.
 
+. Select a dashboard in the *Dashboard* list to see the filtered metrics.
```
Done.

smunje1 commented on modules/odc-monitoring-your-project-metrics.adoc:
```diff
@@ -11,19 +11,25 @@ After you create applications in your project and deploy them, you can use the *
+====
+In the *Dashboard* list, *Kubernetes / Compute Resources / Namespace (Pods)* dashboard is selected by default.
+====
 +
 Use the following options to see further details:
```
I had a meeting with SME @vikram-raj, where he helped me with the demo for this functionality. From that, I can say they are documented properly.

smunje1 commented on modules/odc-monitoring-your-project-metrics.adoc:
```diff
@@ -11,19 +11,25 @@ After you create applications in your project and deploy them, you can use the *
 Use the following options to see further details:
 
-** Select a workload from the *All Workloads* list to see the filtered metrics for the selected workload.
+** Select a dashboard in the *Dashboard* list to see the filtered metrics. All dashboards produce additional sub-menus when selected, except *Kubernetes / Compute Resources / Namespace (Pods)*.
```
Done.

smunje1 commented on modules/odc-monitoring-your-project-metrics.adoc:
```diff
@@ -11,19 +11,25 @@ After you create applications in your project and deploy them, you can use the *
-** Select a workload from the *All Workloads* list to see the filtered metrics for the selected workload.
+** Select a dashboard in the *Dashboard* list to see the filtered metrics. All dashboards produce additional sub-menus when selected, except *Kubernetes / Compute Resources / Namespace (Pods)*.
 ** Select an option from the *Time Range* list to determine the time frame for the data being captured.
+** Set a custom time range by selecting *Custom time range* in the *Time Range* list. You can input or select the *From* and *To* dates and times. Click *Save* to save the custom time range.
```
Done.

smunje1 commented:
Hello @Srivaralakshmi. I have incorporated your feedback. Thank you!
Could you please confirm now?
Updated content:
1) https://drive.google.com/file/d/11Hbc7pbQ1yzSKWJkxd5lguMCGF0ILnL8/view?usp=sharing
2) https://drive.google.com/file/d/1IA76IJraVNlDxURNztMx1zzbe8Z6drAl/view?usp=sharing

Srivaralakshmi commented on modules/odc-monitoring-your-project-metrics.adoc:
```diff
@@ -11,21 +11,34 @@ After you create applications in your project and deploy them, you can use the *
+* Memory usage
+* Bandwidth consumption
+* Network-related information such as the rate of transmitted and received packets and the rate of dropped packets.
++
```
Remove the + in L21

Srivaralakshmi replied:
@smunje1 Refer to this tip in asciiDoc quick reference: https://docs.asciidoctor.org/asciidoc/latest/lists/continuation/#enclose-in-open-block

This will be helpful for future references to too to fix the lists.

Srivaralakshmi commented on modules/odc-monitoring-your-project-metrics.adoc:
```diff
@@ -11,21 +11,34 @@ After you create applications in your project and deploy them, you can use the *
+* Network-related information such as the rate of transmitted and received packets and the rate of dropped packets.
++
+In the *Dashboard* tab, you can access the Kubernetes compute resources dashboards.
++
```
Remove the + in L23

Srivaralakshmi commented on modules/odc-monitoring-your-project-metrics.adoc:
```diff
@@ -11,21 +11,34 @@ After you create applications in your project and deploy them, you can use the *
++
+.Observe dashboard
+image::odc_observe_dashboard.png[]
++
```
Remove the + in L26

Srivaralakshmi commented on modules/odc-monitoring-your-project-metrics.adoc:
```diff
@@ -11,21 +11,34 @@ After you create applications in your project and deploy them, you can use the *
+====
+In the *Dashboard* list, *Kubernetes / Compute Resources / Namespace (Pods)* dashboard is selected by default.
+====
 +
```
Remove the + in L31

Srivaralakshmi commented on modules/odc-monitoring-your-project-metrics.adoc:
```diff
@@ -36,11 +49,17 @@ image::odc_project_metrics.png[]
++
+* See the rules that trigger alerts for the applications in your project.
+* Identify the alerts firing in the project.
+* Silence such alerts if required.
```
Remove the + in L57

Srivaralakshmi commented on modules/odc-monitoring-your-project-metrics.adoc:
```diff
@@ -36,11 +49,17 @@ image::odc_project_metrics.png[]
 +
 .Monitoring alerts
 image::odc_project_alerts.png[]
 +
```
Remove the + in L60

Srivaralakshmi reviewed (changes requested):
@smunje1 Minor changes. PTAL. Thanks!

smunje1 commented:
I have incorporated your feedback as per our discussion on call. Could you please confirm now? Thank you!

Updated content: https://drive.google.com/file/d/1Rsb6-qVETB5HQAuzEyEEFoqPjHTzVCSF/view?usp=sharing

Srivaralakshmi reviewed (approved):
Thanks for the changes @smunje1. Approved!

Srivaralakshmi commented:
/lgtm

smunje1 commented:
Note: As per the discussion with SME @vikram-raj, the Affects Version/s is Openshift 4.9 for this Jira (https://issues.redhat.com/browse/RHDEVDOCS-3111) owing to which, I have raised the PR pointing to 4.11 (main) and we need to cherry-pick it till 4.9.

JStickler commented:
/cherry-pick enterprise-4.9

JStickler commented:
/cherry-pick enterprise-4.10

JStickler commented:
/cherry-pick enterprise-4.11
